src/renderer/components/table/table-cell.tsx

```diff
@@ -20,9 +24,98 @@ export interface TableCellProps extends React.DOMAttributes<HTMLDivElement> {
   _sorting?: Partial<TableSortParams>; // <Table> sorting state, don't use this prop outside (!)
   _sort?(sortBy: TableSortBy): void; // <Table> sort function, don't use this prop outside (!)
   _nowrap?: boolean; // indicator, might come from parent <TableHead>, don't use this prop outside (!)
+
+  /**
+   * Triggers every time this cell is resized by user
+   * @param width desired width of this cell
+   */
+  onResize?(width: number): void;
+
+  /**
+   * Triggers once the user has finished resizing this cell
+   */
+  onResizeComplete?(): void; // triggers a callback when user stops resizing
 }
 
+type ResizeHandlerState = {
+  mousePosX?: number,
+};
+
 export class TableCell extends React.Component<TableCellProps> {
+  private resizeHandlerState?: ResizeHandlerState;
+  private cellContainer: React.RefObject<HTMLDivElement> = React.createRef();
+
+  /**
+   * Stores an instance of `<TableCell>` that is currently responsible for handling resize-related events
+   */
+  private static currentEventHandlingInstance?: TableCell;
```

Comment:
What the point of having `private static` instead of just `private`?

Reply:
So `currentEventHandlingInstance` could be accessed independently from it's own lifecycle (creation/destruction), otherwise it would leak event handlers (will not properly handle listeners removal - event listeners would persist after destruction of a `TableCell` that was handling the events)

Reply:
I'm currently working on integrating a resizing component made by @Nokel81 that would handle this logic on it's own.